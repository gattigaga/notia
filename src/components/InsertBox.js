import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: 480px;
  height: 48px;
  background: white;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.1);
  margin: 32px auto;
  display: flex;
`;

const TextContainer = styled.div`
  flex: 1;
  display: flex;
  cursor: pointer;
  align-items: center;
`;

const Text = styled.span`
  font-style: italic;
  font-size: 14px;
  font-family: Roboto;
  color: #bbb;
  margin-left: 18px;
  user-select: none;
`;

const InsertBox = ({ children, onClick }) => (
  <Container onClick={onClick}>
    <TextContainer>
      <Text>What do you think ?</Text>
    </TextContainer>
    {children}
  </Container>
);

InsertBox.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func
};

export default InsertBox;
