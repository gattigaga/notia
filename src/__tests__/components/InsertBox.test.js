import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import InsertBox from "components/InsertBox";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<InsertBox onClick={onClick} />);
});

afterEach(() => {
  onClick.mockReset();
});

describe("InsertBox", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.find("TextContainer").simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
