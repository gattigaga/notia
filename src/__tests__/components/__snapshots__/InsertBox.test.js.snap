// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsertBox renders correctly 1`] = `
<styled.div>
  <TextContainer
    onClick={[Function]}
  >
    <styled.span>
      What do you think ?
    </styled.span>
  </TextContainer>
</styled.div>
`;
